src/errors.rs

```diff
@@ -10,6 +10,7 @@ pub enum TrieError {
     InvalidData,
     InvalidStateRoot,
     InvalidProof,
+    Invariant(String),
```

Comment:
Hm, I think a rustier thing to do is just panic instead, for true invariants like this.